tools/cli/adminDBScanCommand.go

```diff
@@ -316,6 +316,25 @@ func initializeHistoryManager(c *cli.Context) persistence.HistoryManager {
 	return historyStore
 }
 
+func initializeShardManager(c *cli.Context) persistence.ShardManager {
+	var shardStore persistence.ShardStore
+	dbType := c.String(FlagDBType)
+	logger := loggerimpl.NewNopLogger()
+	switch dbType {
+	case "cassandra":
+		shardStore = initializeCassandraShardStore(c, "current-cluster", logger)
+	default:
+		if sql.PluginRegistered(dbType) {
+			shardStore = initializeSQLShardStore(c, "current-cluster", logger)
+		} else {
+			supportedDBs := append(sql.GetRegisteredPluginNames(), "cassandra")
+			ErrorAndExit(fmt.Sprintf("The DB type is not supported. Options are: %s.", supportedDBs), nil)
+		}
+	}
```

Comment:
Nit: for a better readability, I would check if dbType is valid in the first place, only then load store when we already know that dbType is valid